packages/elements/src/internals/machines/sign-in.machine.ts

```diff
@@ -144,77 +151,145 @@ export const SignInMachine = setup({
         },
       }),
     },
-    'OAUTH.CALLBACK': '.SSOCallbackRunning',
+    'OAUTH.CALLBACK': '#SignIn.SSOCallbackRunning',
   },
+  initial: 'DeterminingState',
   states: {
-    Init: {
+    DeterminingState: {
+      initial: 'Init',
+      states: {
+        Init: {
+          always: [
+            {
+              description: 'If the SignIn resource is empty, invoke the sign-in start flow.',
+              guard: 'isServer',
+              target: 'Server',
+            },
+            {
+              description: 'Wait for the Clerk instance to be ready.',
+              guard: 'isBrowser',
+              target: 'Browser',
+            },
+          ],
+        },
+        Server: {
+          description: 'Determines the state of the sign-in flow on the server. This is a no-op for now.', // TODO: Implement
+          entry: ['debug', log('Server no-op')],
+        },
+        Browser: {
```

Comment:
Why do we need the server/browser awareness? In reality this should only run in the browser 🤔

Reply:
Chatted in-person, but largely kept the flows cleaner and safer as the machine can run server-side.